static/app/components/modals/demoEndModal.tsx

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {fetchGuides} from 'sentry/actionCreators/guides';
+import {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {navigateTo} from 'sentry/actionCreators/navigation';
+import {Client} from 'sentry/api';
+import Button from 'sentry/components/button';
+import ModalTask from 'sentry/components/onboardingWizard/modalTask';
+import {SidebarPanelKey} from 'sentry/components/sidebar/types';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t, tct} from 'sentry/locale';
+import SidebarPanelStore from 'sentry/stores/sidebarPanelStore';
+import {useRouteContext} from 'sentry/utils/useRouteContext';
+
+type Props = ModalRenderProps & {orgSlug: string | null; tour: string};
+
+function DemoEndingModal({tour, closeModal, orgSlug}: Props) {
+  const api = new Client();
+
+  const routeContext = useRouteContext();
+  const {router} = routeContext;
+
+  function togglePanel(panel: SidebarPanelKey) {
+    SidebarPanelStore.togglePanel(panel);
+  }
+
+  let cardTitle, body, guides, location;
+  switch (tour) {
+    case 'issues':
+      cardTitle = 'Issues Tour';
+      body = 'completing the Issues tour';
+      guides = ['issues_v3', 'issue_stream_v3'];
+      location = `/organizations/${orgSlug}/issues/`;
+      break;
+    case 'performance':
+      cardTitle = 'Performance Tour';
+      body = 'completing the Performance tour';
+      guides = ['performance', 'transaction_summary', 'transaction_details_v2'];
+      location = `/organizations/${orgSlug}/performance/`;
+      break;
+    case 'releases':
+      cardTitle = 'Releases Tour';
+      body = 'completing the Releases tour';
+      guides = ['releases_v2', 'react-native-release', 'release-details_v2'];
+      location = `/organizations/${orgSlug}/releases/`;
+      break;
+    case 'tabs':
+      cardTitle = 'Check out the different tabs';
+      body = 'checking out the different tabs';
+      guides = ['sidebar_v2'];
+      location = `/organizations/${orgSlug}/projects/`;
+      break;
+    default:
+  }
+
+  const sandboxData = window.SandboxData;
+  const url = sandboxData?.cta?.url || 'https://sentry.io/signup/';
+
+  async function handleRestart() {
+    await Promise.all(
+      guides.map(guide =>
+        api.requestPromise('/assistant/', {
+          method: 'PUT',
+          data: {guide, status: 'restart'},
+        })
+      )
+    ).then(() => {
+      if (tour === 'issues') {
+        localStorage.setItem('issueGuide', '1');
+      }
+      if (tour === 'tabs') {
+        localStorage.setItem('sidebarGuide', '1');
+      }
+
+      if (closeModal) {
+        closeModal();
+      }
```

Comment:
```suggestion
      closeModal?.();
```

you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here